Buildings/Experimental/DHC/CentralPlants/Cooling/Subsystems/CoolingTowerWithBypass.mo

```diff
@@ -0,0 +1,397 @@
+  Modelica.Blocks.Interfaces.RealOutput PFan[num](
+    each final quantity="Power",
+    each final unit="W")
+    "Electric power consumed by fan"
+    annotation (Placement(transformation(extent={{100,50},{120,70}})));
+  Modelica.Blocks.Interfaces.RealOutput TLvg[num](
+    each final unit="K",
+    each displayUnit="degC")
+    "Leaving water temperature"
+    annotation (Placement(transformation(extent={{100,20},{120,40}})));
+  Buildings.Experimental.DHC.CentralPlants.Cooling.Subsystems.CoolingTowerParellel cooTowSys(
+    final use_inputFilter=use_inputFilter,
+    redeclare final package Medium=Medium,
+    final num=num,
+    final show_T=show_T,
+    final m_flow_nominal=m_flow_nominal/num,
+    final dp_nominal=dp_nominal,
+    final ratWatAir_nominal=ratWatAir_nominal,
+    final TAirInWB_nominal=TAirInWB_nominal,
+    final TWatIn_nominal=TWatIn_nominal,
+    final dT_nominal=dT_nominal,
+    final PFan_nominal=PFan_nominal,
+    final energyDynamics=energyDynamics)
+    "Cooling tower system"
+    annotation (Placement(transformation(extent={{-10,-10},{10,10}})));
+  Buildings.Fluid.Actuators.Valves.TwoWayEqualPercentage valByp(
+    redeclare final package Medium=Medium,
+    final m_flow_nominal=m_flow_nominal*0.0001,
+    final dpValve_nominal=dp_nominal,
+    final use_inputFilter=use_inputFilter)
+    "Condenser water bypass valve"
+    annotation (Placement(transformation(extent={{-10,-10},{10,10}},origin={0,-40})));
+  Buildings.Fluid.Sensors.TemperatureTwoPort senTCWSup(
+    redeclare final package Medium=Medium,
+    final m_flow_nominal=m_flow_nominal,
+    final T_start=Medium.T_default)
+    "Temperature sensor"
+    annotation (Placement(transformation(extent={{60,10},{80,-10}})));
+  Modelica.Blocks.Sources.Constant TSetByPas(
+    final k=TMin)
+    "Bypass loop temperature setpoint"
+    annotation (Placement(transformation(extent={{-90,-60},{-70,-40}})));
+  Buildings.Controls.Continuous.LimPID bypValCon(
+    u_s(
+      final unit="K",
+      displayUnit="degC"),
+    u_m(
+      final unit="K",
+      displayUnit="degC"),
+    controllerType=Modelica.Blocks.Types.SimpleController.PI,
+    k=1,
+    Ti=60,
+    final reset=Buildings.Types.Reset.Parameter,
+    final y_reset=0)
+    "Bypass valve controller"
+    annotation (Placement(transformation(extent={{-60,-60},{-40,-40}})));
+  Buildings.Controls.Continuous.LimPID cooTowSpeCon(
+    u_s(
+      final unit="K",
+      displayUnit="degC"),
+    u_m(
+      final unit="K",
+      displayUnit="degC"),
+    final reverseActing=false,
+    final controllerType=controllerType,
+    final k=k,
+    final Ti=Ti)
+    "Cooling tower fan speed controller"
+    annotation (Placement(transformation(extent={{-12,50},{8,70}})));
+  Buildings.Controls.OBC.CDL.Logical.Switch swi
+    "Output the input of higher value"
+    annotation (Placement(transformation(extent={{-40,50},{-20,70}})));
+  Buildings.Controls.OBC.CDL.Continuous.Greater gre
```

Comment:
This should be a hysteresis. See https://simulationresearch.lbl.gov/modelica/userGuide/bestPractice.html#controls